src/tfr_mission_control/src/tfr_mission_control/mission_control.cpp

```diff
@@ -0,0 +1,409 @@
+                [this] () {performTeleop(tfr_utilities::TeleopCode::RIGHT);});
+        connect(ui.right_button,&QPushButton::released,
+                [this] () {performTeleop(tfr_utilities::TeleopCode::STOP_DRIVEBASE);});
+
+        //set upp our action servers
+        ROS_INFO("Mission Control: connecting autonomy");
+        autonomy.waitForServer();
+        ROS_INFO("Mission Control: connected autonomy");
+        ROS_INFO("Mission Control: connecting teleop");
+        teleop.waitForServer();
+        ROS_INFO("Mission Control: connected teleop");
+    }
+
+    /*
+     * This get's called before the destructor, and apparently we need to
+     * deallocate our ros objects manually here according to the documentation.
+     * Publishers and subscribers especially
+     * */
+    void MissionControl::shutdownPlugin()
+    {
+        //note because qt plugins are weird we need to manually kill ros entities
+        com.shutdown();
+        autonomy.cancelAllGoals();
+        autonomy.stopTrackingGoal();
+        teleop.cancelAllGoals();
+        teleop.stopTrackingGoal();
+    }
+
+    /* ========================================================================== */
+    /* Settings                                                                   */
+    /* ========================================================================== */
+
+    /* we inherit this, but I don't think we need it*/ 
+    void MissionControl::saveSettings(
+            qt_gui_cpp::Settings& plugin_settings,
+            qt_gui_cpp::Settings& instance_settings) const
+    {
+    }
+
+    /* we inherit this, but I don't think we need it*/ 
+    void MissionControl::restoreSettings(
+            const qt_gui_cpp::Settings& plugin_settings,
+            const qt_gui_cpp::Settings& instance_settings)
+    {
+    }
+
+    /* ========================================================================== */
+    /* Methods                                                                    */
+    /* ========================================================================== */
+    
+    /* greys/ungreys all teleop buttons, and tell's system whether to process teleop or
+     * not
+     * */
+    void MissionControl::setTeleop(bool value)
+    {
+        ui.left_button->setEnabled(value);
+        ui.right_button->setEnabled(value);
+        ui.forward_button->setEnabled(value);
+        ui.backward_button->setEnabled(value);
+        ui.cw_button->setEnabled(value);
+        ui.ccw_button->setEnabled(value);
+        ui.reset_starting_button->setEnabled(value);
+        ui.reset_dumping_button->setEnabled(value);
+        ui.autonomy_button->setEnabled(value);
+        ui.dump_button->setEnabled(value);
+        ui.dig_button->setEnabled(value);
+        teleopEnabled = value;
+    }
+
+    /*
+     * Toggles the autonomy stop button
+     * */
+    void MissionControl::setAutonomy(bool value)
+    {
+        ui.teleop_button->setEnabled(value);
```

Comment:
The naming here is really confusing, just because I'm seeing "set autonomy" maps to "enable the teleop button", which feels really backwards to me. It just worries me that something's gonna get flipped, so I'd like to see a bit more documentation/a name change to make this clearer.

Reply:
Adam it's a window state, I think it makes sense. The only button you should be able to press when you are in autonomous mode is the enable teleop.